frontend/__tests__/src/pages/Chapters.test.tsx

```diff
@@ -39,6 +39,22 @@ describe('ChaptersPage Component', () => {
     })
   })
 
+  test('renders SearchBar and data concurrently', async () => {
```

Comment:
Does it check whether search and pagination are not rendered before data is loaded?

My understanding it checks for the loading indicator (we already have such test) and then it checks that search is present.

Could you add the requested checks too?

Thank you!

Reply:
I am making some modifications 


as when data is only rendered when it is available in the frontend so in this test search and data are loaded concurrently if the data is not there the search component will also be not there .

for now I am also going to add same thing with pagination 


Which will ensure all three are rendered concurrently only when data is present.

Reply:
The already added tests are for specific cases like particular to pagination or being particular to render data .

I guess they aren't checking the concurrent appearance of the components.

Reply:
Let me check once again !

Reply:
@arkid15r 

The recent changes in the tests ensure that the search bar and pagination components are not rendered while the loader is visible initially over the page. Only after the data is successfully fetched, are the pagination and search bar rendered which is checked in the second expect block !